### ISSUE TITLE
Overly restrictive name constraints

### ISSUE THREAD

**Issue by FooChao** [FooChao/pe#4]
![image.png](https://raw.githubusercontent.com/FooChao/pe/main/files/2a5beed3-2c79-4744-8ed6-45e5f57d7b9f.png)

![image.png](https://raw.githubusercontent.com/FooChao/pe/main/files/fc957ae0-ed4b-4d20-9146-fedbc1b5c307.png)

Names with `S/O`, `D/O`, `-`, or `’` are valid names, however they are not allowed by the address book.
E.g. 
John-Henry
Rajesh Kumar S/O Mahesh Sharma
Priya Sharma D/O Mahesh Sharma
O'Neil

**Comment by nus-pe-script** [comment]
# Team's Response
No details provided by team.

## The 'Original' Bug

[The team marked this bug as a duplicate of the following bug]

># Overzealous handling of inputs for names
>
>Does not allow for special characters in names. In Singapore, 's/o' is common a prefix in people's names, but this app does not allow for '/' within names
>
>**Steps to Reproduce:**
>1. edit 11 n/Michael s/o Kumar
>
>**Expected Behaviour:**
>I should be able to add names with '/', especially considering the prevalence of it in Singapore.
>
>**Observed Behaviour:**
>I am unable to add such names.
>
>**Suggested Improvements:**
>Allow me to add names with '/'.
>
>![Screenshot 2024-11-15 at 16.23.31.png](https://raw.githubusercontent.com/nus-cs2103-AY2425S1/pe/master/files/202b59ba-f4ca-4050-82ff-d4f827529735.png)
>
>
>
>
>
>
><hr>
><sub>[original: nus-cs2103-AY2425S1/pe-interim#2405] [original labels: severity.Medium type.FeatureFlaw]</sub>
>
>
>

## Their Response to the 'Original' Bug

[This is the team's response to the above 'original' bug]

>![image.png](https://raw.githubusercontent.com/nus-cs2103-AY2425S1/pe/master/files/8eebe470-c4c2-4224-a7a3-f949d363bead.png)
>
>
>We do recognise this issue, but firstly, this has been mentioned in the UG, secondly this would not affect users much as the full government name is not required in as defined in our readme, and finally, we have provided workarounds for the user in the user guide.
# Items for the Tester to Verify
## :question: Issue duplicate status

Team chose to mark this issue as a duplicate of another issue (as explained in the _**Team's response**_ above)

- [ ] I disagree

**Reason for disagreement:** [replace this with your explanation]

<catcher-end-of-segment><hr>
## :question: Issue response

Team chose [`response.NotInScope`]

- [x] I disagree

**Reason for disagreement:** I disagree that it is out of scope for the following 9 reasons:

1. Firstly, as mentioned in my original report, the dev team clearly stated `FULL NAME` in their UG which they clearly choose to reply to the other similar response because the other tester did not notice it.

![image.png](https://raw.githubusercontent.com/FooChao/pe/main/files/c0e15f35-a6d6-4c5a-babd-03bb45ed0d30.png)

To emphasise it, they even mention it twice in their UG, once in the command format, and another in the constraints point form below.


![image.png](https://raw.githubusercontent.com/FooChao/pe/main/files/de4bd5c4-a182-4f6b-a930-bdc883c98f07.png)


To repeat, it is impossible to type in full name without `S/O` or `D/O` or  `-` or `'`.

According to CS2103T website,


![image.png](https://raw.githubusercontent.com/FooChao/pe/main/files/2e51f15f-32dd-48ea-9eea-1c96070b28bd.png)

This can still be considered a feature flaw which is within scope as users are inconvenienced because the UG clearly tell us to use full name which clearly requires `S/O` or `D/O` or `-` or `'`.


2. Secondly, even if, UG did not tell us to use full name (which they clearly did), as a real estate agent (target audience stated clearly in their website landing page or README), it is important to know their legal name because we have to write it in their contract. It is clearly stated in their README which follow-ups are also part of their purpose i.e contract writing is also part of it.


![image.png](https://raw.githubusercontent.com/FooChao/pe/main/files/3a424911-db64-4694-bb81-0b55a675f64f.png)

![image.png](https://raw.githubusercontent.com/FooChao/pe/main/files/c52932e1-ef7f-4f21-855e-c2d584d2b00a.png)


(note: target audience can't be found in their user guide because they forgot to update it in their UG but real estate agent is their target audience as shown in everywhere else)

3. Thirdly, regarding them 'mentioning' it in UG. Firstly, if it inconvenience users, it inconvenience users, asking them to write out `son of` to replace `S/O` is not a fair decision to users because of point 1 and 2. Moreover, UG did not explicitly tell them it is a replacement, it just say what is allowed what is not.

4. Fourthly, even without considering first 3 points in my issue report, they did not give alternative for `D/O` , `-` or `'`

Using my original example , they cannot expect use to type `John-Henry` as `John hyphen Henry`. It makes no sense and they never give us an alternative anywhere in the whole of UG.

5. Fifthly, they claimed that  it is stated in the README that full government name is not needed. I went to check it is not there. You may verify it by going to the link below. No screenshots taken because their README is too long.

https://github.com/AY2425S1-CS2103-F09-2/tp

6. Sixthly, even if they did mention in README (which they clearly did not), testers are clearly asked to not used github unless we got legit reason, i.e we have no access to their README. Extract of CS2103T admin instructions shown below. As such, even if it is in their README, nobody cares if nobody can see it. I will refer to UG and use `FULL NAME` as stated in their UG.


![image.png](https://raw.githubusercontent.com/FooChao/pe/main/files/2b1c9633-f50e-463d-b492-796cc53d28d2.png)

How are we supposed to know we supposed to check their README?

We are also asked to gauge their product using UG and not README.

7. Seventhly, as mentioned by the other tester, `In Singapore, 's/o' is common a prefix in people's names`


8. Eighthly, changing this flaw only change one line of code of String Regex in name class. According to CS2103T website,


![image.png](https://raw.githubusercontent.com/FooChao/pe/main/files/817b3294-5c84-4c90-a2d2-3aff48c4b260.png)

This flaw does not qualify for feature flaw because it match point 2. Justification as such.

Can it be implemented to work in a different better way? Yes, as stated in point 1-4 and 7, able to add `S/O`, `D/O`, `'`, `-` is highly important considering real estate agent needs to write contract and UG clearly asked us to use full name and how common it is.


Can it be changed easily? Definitely, String Regex is only 1 line of code as shown in the screenshot of their code below. (ignore the typo that says address it is default ab3 typo and it is from name class)


![image.png](https://raw.githubusercontent.com/FooChao/pe/main/files/1e1e5e14-c0de-4429-bb5f-ac6b57b059e2.png)

As seen below, changing String REGEX is just one line of code. Considering the amount of inconvenience it brings and how easy is it to fix, it should never ever qualify for not in scope.

9. The most ironic and iconic reason of why this should be accepted is this.

Drumroll please.

![image.png](https://raw.githubusercontent.com/FooChao/pe/main/files/863a64b8-1979-479c-afb4-21908c0648ea.png)

They accepted a bug which is clearly same as mine or the one reported as duplicate as mine but they did not put mine/the other one as duplicate of it. Instead they NotInScope me which is weird and controversial.

https://github.com/nus-cs2103-AY2425S1/pe-dev-response/issues/2058

Link above.

As such even their group mate feel that it is valid but another group mate wanted to find excuse that it is not in scope.

Fun fact: I even tell them but opening an issue on their tp's repo to inform them that mine is a duplicate and in the event where i dispute it they will get penalised twice because i do not want to write this long essay and I want to focus on exam but they ignore me.
(I am not allowed to file duplicate for them right?) 


In conclusion, I feel that this issue should not be not in scope because it is common for names to have these symbols, their target audience needs to use full name for the job scope, their UG mention it should be full name and it only takes one line of code to change it. Moreover, all their counter-arguments are either flawed or is a lie in the first place(the part about README).

<catcher-end-of-segment><hr>
